src/popupmenu.c

```diff
@@ -416,6 +416,71 @@ pum_under_menu(int row, int col, int only_redrawing)
 	    && col < pum_col + pum_width + pum_scrollbar;
 }
 
+/*
+ * displays text on the popup menu with specific attributes.
+ */
+    static void
+pum_screen_put_with_attr(int row, int col, char_u *text, int textlen, int attr)
```

Comment:
can you make this interface similar to `screen_puts_len()`? 

`pum_screen_put_with_attr(text, textlen, row, col, attr)`